lib/ex_doc/retriever.ex

```diff
@@ -145,60 +140,58 @@ defmodule ExDoc.Retriever do
       doc_line: doc_line,
       typespecs: Enum.sort_by(types, &{&1.name, &1.arity}),
       source_path: source_path,
-      source_url: source_link(source, line)
+      source_url: source_link(source, line),
+      language: module_data.language
     }
 
     put_in(node.group, GroupMatcher.match_module(config.groups_for_modules, node))
   end
 
+  defp nesting_info(title, prefixes) do
+    prefixes
+    |> Enum.find(&String.starts_with?(title, &1 <> "."))
+    |> case do
+      nil -> {nil, nil}
+      prefix -> {String.trim_leading(title, prefix <> "."), prefix}
+    end
+  end
+
   defp sort_key(name, arity) do
     first = name |> Atom.to_charlist() |> hd()
     {first in ?a..?z, name, arity}
   end
 
-  defp doc_ast(format, %{"en" => doc_content}, options),
-    do: DocAST.parse!(doc_content, format, options)
+  defp doc_ast(format, %{"en" => doc_content}, options) do
+    DocAST.parse!(doc_content, format, options)
+  end
 
-  defp doc_ast(_, _, _options),
-    do: nil
+  defp doc_ast(_, _, _options) do
+    nil
+  end
 
   # Module Helpers
 
   defp get_module_data(module, docs_chunk) do
+    {:docs_v1, _, language, _, _, _, _} = docs_chunk
+    language = ExDoc.Language.get(language)
+    extra = language.module_data(module)
+
     %{
+      id: extra.id,
+      title: extra.title,
       name: module,
-      type: get_type(module),
+      language: language,
+      type: extra.type,
+      skip: extra.skip,
       specs: get_specs(module),
       impls: get_impls(module),
       callbacks: get_callbacks(module),
       abst_code: get_abstract_code(module),
```

Comment:
We should probably move this to the language callbacks... i think Elixir and Erlang are the only two languages that benefit from looking at the abstract code. LFE, Gleam, etc either won't have it or it will be inaccurate.